staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -299,6 +308,10 @@ func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store R
 		r.UseWatchList = ptr.To(clientfeatures.FeatureGates().Enabled(clientfeatures.WatchListClient))
 	}
 
+	if r.metrics == nil {
+		r.metrics = newReflectorMetrics(r.name, reflect.TypeOf(expectedType).Elem().String(), options.MetricsProvider)
```

Comment:
> The informer and reflector provider should probably be exposed in their respective options to allow consumers to override the metrics

@dgrisonnet ReflectorMetricsProvider has already been provided in `ReflectorOptions`.